src/libraries/Microsoft.Extensions.Configuration.Binder/gen/Emitter/CoreBindingHelpers.cs

```diff
@@ -718,6 +719,16 @@ private void EmitBindingLogicForEnumerableWithAdd(TypeRef elementTypeRef, string
                 EmitEndBlock();
             }
 
+            // EmitBindCheckForSectionValue produce the following code:
+            // if (!string.IsNullOrEmpty(section.Value) && !section.GetChildren().GetEnumerator().MoveNext()) { continue; }
+            private void EmitBindCheckForSectionValue()
+            {
+                // We utilize GetEnumerator().MoveNext() instead of employing Linq's Any() since there is no assurance that the System.Linq reference is included.
+                EmitStartBlock($"if (!string.IsNullOrEmpty({Expression.sectionValue}) && !{Identifier.section}.{Identifier.GetChildren}().{Identifier.GetEnumerator}().{Identifier.MoveNext}())");
```

Comment:
Maybe this is what you meant? 

https://github.com/dotnet/runtime/blob/bfa39b303ab33306b66ec24c72ccc8fc4fe17bb2/src/libraries/Microsoft.Extensions.Configuration.Binder/gen/Emitter/CoreBindingHelpers.cs#L1186

Reply:
Per offline discussion, we might be able to reuse some of the existing helpers, or follow the established pattern.

Reply:
Talked offline with @ericstj:
After looking, it is not worth using any other helpers because the helpers emitted in a specific condition and we want to avoid creating dependencies to ensure the helper is always emitted in our case. The check we already have is simple enough that doesn't need helpers.

Reply:
Your call on this suggestion.  It doesn't impact correctness and I can see how it might churn the generated code more than we want.